packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -292,7 +326,14 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
 
     // Do a bit of contortions to only pass the `DisableRollback` flag if it's true. That way,
     // CloudFormation won't balk at the unrecognized option in regions where the feature is not available yet.
-    const disableRollback = options.rollback === false ? { DisableRollback: true } : undefined;
+    let disableRollback;
+
+    if (options.rollback === undefined && options.hotswap === true) {
+      disableRollback = { DisableRollback: true };
+    } else {
+      disableRollback = options.rollback === false ? { DisableRollback: true } : undefined;
+    }
```

Comment:
We're repeating the `{ DisableRollback: true }` expression for no reason here.

```suggestion
    const shouldDisableRollback = (ptions.rollback === undefined && options.hotswap === true) || options.rollback === false;
    const disableRollback = shouldDisableRollback ? { DisableRollback: true } : undefined;
```